server/gitlab/api.go

```diff
@@ -11,16 +11,198 @@ import (
+		URL:                      &webhookOptions.URL,
+		ConfidentialNoteEvents:   &webhookOptions.ConfidentialNoteEvents,
+		PushEvents:               &webhookOptions.PushEvents,
+		IssuesEvents:             &webhookOptions.IssuesEvents,
+		ConfidentialIssuesEvents: &webhookOptions.ConfidentialIssuesEvents,
+		MergeRequestsEvents:      &webhookOptions.MergeRequestsEvents,
+		TagPushEvents:            &webhookOptions.TagPushEvents,
+		NoteEvents:               &webhookOptions.NoteEvents,
+		JobEvents:                &webhookOptions.JobEvents,
+		PipelineEvents:           &webhookOptions.PipelineEvents,
+		WikiPageEvents:           &webhookOptions.WikiPageEvents,
+		EnableSSLVerification:    &webhookOptions.EnableSSLVerification,
+		Token:                    &webhookOptions.Token,
+	}
+
+	groupHook, _, err := client.Groups.AddGroupHook(group.ID, &groupHookOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	groupHookInfo := getGroupHookInfo(groupHook)
+
+	return groupHookInfo, nil
+}
+
+// NewProjectHook creates a webhook associated with a GitLab project
+func (g *gitlab) NewProjectHook(user *GitlabUserInfo, projectID interface{}, webhookOptions *AddWebhookOptions) (*WebhookInfo, error) {
+	client, err := g.gitlabConnect(*user.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	projectHookOptions := internGitlab.AddProjectHookOptions{
+		URL:                      &webhookOptions.URL,
+		ConfidentialNoteEvents:   &webhookOptions.ConfidentialNoteEvents,
+		PushEvents:               &webhookOptions.PushEvents,
+		IssuesEvents:             &webhookOptions.IssuesEvents,
+		ConfidentialIssuesEvents: &webhookOptions.ConfidentialIssuesEvents,
+		MergeRequestsEvents:      &webhookOptions.MergeRequestsEvents,
+		TagPushEvents:            &webhookOptions.TagPushEvents,
+		NoteEvents:               &webhookOptions.NoteEvents,
+		JobEvents:                &webhookOptions.JobEvents,
+		PipelineEvents:           &webhookOptions.PipelineEvents,
+		WikiPageEvents:           &webhookOptions.WikiPageEvents,
+		EnableSSLVerification:    &webhookOptions.EnableSSLVerification,
+		Token:                    &webhookOptions.Token,
+	}
+
+	projectHook, _, err := client.Projects.AddProjectHook(projectID, &projectHookOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	projectHookInfo := getProjectHookInfo(projectHook)
+
+	return projectHookInfo, nil
+}
+
+// GetGroupHooks gathers all the group level hooks for a GitLab group.
+func (g *gitlab) GetGroupHooks(user *GitlabUserInfo, owner string) ([]*WebhookInfo, error) {
+	client, err := g.gitlabConnect(*user.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	hooks, _, err := client.Groups.ListGroupHooks(owner)
+	if err != nil {
+		return nil, err
+	}
+
+	var webhooks []*WebhookInfo
+	for _, hook := range hooks {
+		webhooks = append(webhooks, getGroupHookInfo(hook))
+	}
+
+	return webhooks, nil
+}
+
+//Stringify produces a multiline bulleted string for displaying webhook information.
+func (w *WebhookInfo) Stringify() string {
```

Comment:
`fmt` definer a `Stringer` interface with one only method [`String()`](https://pkg.go.dev/fmt?tab=doc#Stringer). Does it make sense to follow the idiomatic naming here?

Reply:
My thinking on this one was that it was a formated string and stringify conveyed that.